survey_mailer/survey_mailer.go

```diff
@@ -0,0 +1,204 @@
+		EventNameQuery: "%chapter meeting%",
+		SurveySent:     "0",
+	})
+	if err != nil {
+		log.Printf("Failed to get today's chapter meeting events: %v", err)
+		return
+	}
+
+	// Iterate through events
+	for _, event := range events {
+
+		// Set email subject & body
+		subject := "Survey: " + event.EventName
+		eventDateParam := event.EventDate.Format("2006-01-02")
+		bodyText := `Thank you for attending the chapter meeting! Please take this quick survey: https://docs.google.com/forms/d/e/1FAIpQLSfc_mgwH_zYYEQ5MTJwgyvCy5klsY_xrVBXgTDHM8sSxLIJrQ/viewform?usp=pp_url&entry.502269384=` + eventDateParam
+		bodyHtml := `<p>Thank you for attending the chapter meeting! Please <a href="https://docs.google.com/forms/d/e/1FAIpQLSfc_mgwH_zYYEQ5MTJwgyvCy5klsY_xrVBXgTDHM8sSxLIJrQ/viewform?usp=pp_url&entry.502269384=` + eventDateParam + `">click here</a> to take a quick survey.</p>`
+
+		log.Println("Sending chapter meeting survey for event:", event.EventName)
+
+		// send all emails, including "missing" email
+		bulkSendEmails(event, subject, bodyText, bodyHtml)
+
+		// updateSurveyStatus
+		updateSurveyStatus(db, event.ID)
+	}
+}
+
+func surveyProtest(db *sqlx.DB, queryDate string) {
+
+	log.Println("Looking for protest events on", queryDate)
+
+	// Get yesterday's protest events that haven't had surveys sent yet
+	events, err := model.GetEvents(db, model.GetEventOptions{
+		DateFrom:   queryDate,
+		DateTo:     queryDate,
+		EventType:  "%action",
+		SurveySent: "0",
+	})
+
+	if err != nil {
+		log.Printf("Failed to get today's protest events: %v", err)
+		return
+	}
+
+	// Iterate through events
+	for _, event := range events {
+
+		// Set email subject & body
+		subject := "Survey: " + event.EventName
+		eventNameParam := strings.Replace(event.EventName, " ", "+", -1)
+		bodyText := `Thank you for taking part in direct action! Please take this quick survey: https://docs.google.com/forms/d/e/1FAIpQLScfrPtPxmYAroODhBkwUGq753JPykYKNdosg4gUR_SRng8BRQ/viewform?usp=pp_url&entry.466557185=` + eventNameParam + ". If you captured any photos or videos, please upload them here: dxe.io/upload."
+		bodyHtml := `<p>Thank you for taking part in direct action! Please <a href="https://docs.google.com/forms/d/e/1FAIpQLScfrPtPxmYAroODhBkwUGq753JPykYKNdosg4gUR_SRng8BRQ/viewform?usp=pp_url&entry.466557185=` + eventNameParam + `">click here</a> to take a quick survey.</p><p>If you captured any photos or videos, please upload them <a href="http://dxe.io/upload">here</a>.</p>`
+
+		log.Println("Sending protest survey for event:", event.EventName)
+
+		// send all emails, including "missing" email
+		bulkSendEmails(event, subject, bodyText, bodyHtml)
+
+		// updateSurveyStatus
+		updateSurveyStatus(db, event.ID)
+	}
+}
+
+func surveyMailerWrapper(db *sqlx.DB) {
+	defer func() {
+		if r := recover(); r != nil {
+			log.Println("Recovered from panic in survey mailer", r)
+		}
+	}()
+
+	// Get current hour of day & current day of week
+	weekday := time.Now().Weekday()
```

Comment:
nit: I'd do:

```
now := time.Now()
```

and change all of these to reuse that same now value. Otherwise, you can have weird race conditions if this function happens to get called at a time boundary.